dotCMS/src/main/java/com/dotcms/keyvalue/business/KeyValueAPI.java

```diff
@@ -0,0 +1,141 @@
+package com.dotcms.keyvalue.business;
+
+import java.util.List;
+
+import com.dotcms.contenttype.model.type.BaseContentType;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.contenttype.model.type.KeyValueContentType;
+import com.dotcms.keyvalue.model.DefaultKeyValue;
+import com.dotcms.keyvalue.model.KeyValue;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+
+/**
+ * Provides access to Key/Value contents in the system. This API provides easy access to any kind of
+ * Content Type based on a key/value data structure, such as the Language Variables.
+ * 
+ * @author Jose Castro
+ * @version 4.2.0
+ * @since Jun 19, 2017
+ *
+ */
+public interface KeyValueAPI {
+
+    /**
+     * Transforms a given {@link Contentlet} object into a {@link KeyValue} object.
+     * 
+     * @param contentlet - The Contentlet containing the Key/Value data.
+     * @return The transformed {@link KeyValue} object.
+     */
+    public default KeyValue fromContentlet(final Contentlet contentlet) {
+        if (null == contentlet) {
+            throw new DotStateException("The contentlet cannot be null.");
+        }
+        try {
+            if (!contentlet.getContentType().baseType().equals(BaseContentType.KEY_VALUE)) {
+                throw new DotStateException(String.format("The contentlet with ID '%s' is not a KeyValue content.",
+                                contentlet.getIdentifier()));
+            }
+        } catch (DotSecurityException | DotDataException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)